pkg/codegen/nodejs/gen_program.go

```diff
@@ -941,7 +941,32 @@ func (g *generator) genResourceDeclaration(w io.Writer, r *pcl.Resource, needsDe
 	if r.Options != nil && r.Options.Range != nil {
 		rangeType := r.Options.Range.Type()
 		rangeExpr := r.Options.Range
-		if model.ContainsOutputs(r.Options.Range.Type()) {
+		if g.isComponent {
+			// In the case of a component, the range expression must treat Inputs as
```

Comment:
I really don't think this is component specific, anytime we have input types (i.e T|Output[T]) we need to be converting them correctly by unifying on Output[T].